optimum/neuron/modeling_base.py

```diff
@@ -0,0 +1,440 @@
+        **kwargs_shapes,
+    ) -> "NeuronModel":
+        """
+        Export a vanilla Transformers model into a neuron-compiled TorchScript Module using `optimum.exporters.neuron.export`.
+
+        Arguments:
+            kwargs_shapes (`Dict`):
+                Shapes to use during inference. This argument allows to override the default shapes used during the export.
+        """
+        if task is None:
+            task = cls._auto_model_to_task(cls.auto_model_class)
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        model = TasksManager.get_model_from_task(
+            task=task,
+            model_name_or_path=model_id,
+            subfolder=subfolder,
+            revision=revision,
+            framework="pt",
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+        )
+
+        neuron_config_constructor = TasksManager.get_exporter_config_constructor(
+            model=model, exporter="neuron", task=task
+        )
+
+        input_shapes = {}
+        for name in neuron_config_constructor.func.get_mandatory_axes_for_task(task):
+            static_shape = kwargs_shapes.get(name, None) or getattr(config, "neuron_" + name, None)
+            if static_shape is None:
+                raise AttributeError(
+                    f"Cannot find the value of `{name}` from arguments nor the `config`. `{name}` is mandatory"
+                    " for exporting the model to the neuron format, please set the value explicitly."
+                )
+            else:
+                input_shapes[name] = static_shape
+        if is_neuron_available() and dynamic_batch_size is True and "batch_size" in input_shapes:
+            input_shapes["batch_size"] = 1
+        neuron_config = neuron_config_constructor(model.config, dynamic_batch_size=dynamic_batch_size, **input_shapes)
+
+        # Get compilation arguments
+        auto_cast_type = None if auto_cast is None else auto_cast_type
+        compiler_kwargs = {
+            "auto_cast": auto_cast,
+            "auto_cast_type": auto_cast_type,
+            "disable_fast_relayout": disable_fast_relayout,
+        }
+
+        input_names, output_names = export(
+            model=model,
+            config=neuron_config,
+            output=save_dir_path / NEURON_FILE_NAME,
+            **compiler_kwargs,
+        )
+
+        store_compilation_config(config, input_shapes, compiler_kwargs, input_names, output_names, dynamic_batch_size)
+
+        config.save_pretrained(save_dir_path)
+        maybe_save_preprocessors(model_id, save_dir_path, src_subfolder=subfolder)
+
+        return cls._from_pretrained(save_dir_path, config, model_save_dir=save_dir, neuron_config=neuron_config)
+
+    def forward(self, *args, **kwargs):
+        raise NotImplementedError
```

Comment:
Make it an abtract method?

Reply:
Actually I shall remove this as it is already an abstract method in `OptimizedModel`.